### PR TITLE
Reduce deployment replicas in e2e tests

### PR DESCRIPTION
**Pull request by 0xmichalis**
Fixes https://github.com/kubernetes/kubernetes/issues/41063

There are still two tests that run multiple replicas (testScaledRolloutDeployment, testIterativeDeployments), we may want to revisit them at some point.

### CODE REVIEW

**Comment by kargakis** [comment]
@kubernetes/sig-apps-test-failures

File: test/e2e/deployment.go [hunk 1]
```diff
@@ -437,14 +437,14 @@ func testRolloverDeployment(f *framework.Framework) {
...
-	deploymentReplicas := int32(3)
+	deploymentReplicas := int32(1)
 	deploymentImage := "gcr.io/google_samples/gb-redisslave:nonexistent"
 	deploymentStrategyType := extensions.RollingUpdateDeploymentStrategyType
 	framework.Logf("Creating deployment %q", deploymentName)
 	newDeployment := framework.NewDeployment(deploymentName, deploymentReplicas, deploymentPodLabels, deploymentImageName, deploymentImage, deploymentStrategyType)
 	newDeployment.Spec.Strategy.RollingUpdate = &extensions.RollingUpdateDeployment{
-		MaxUnavailable: func(i int) *intstr.IntOrString { x := intstr.FromInt(i); return &x }(1),
-		MaxSurge:       func(i int) *intstr.IntOrString { x := intstr.FromInt(i); return &x }(1),
+		MaxUnavailable: intOrStrP(0),
```
Comment (mfojtik) [inline test/e2e/deployment.go]: why this change?

File: test/e2e/deployment.go [hunk 2]
```diff
@@ -483,8 +490,19 @@ func testRolloverDeployment(f *framework.Framework) {
 	framework.Logf("Make sure deployment %q is complete", deploymentName)
 	err = framework.WaitForDeploymentStatus(c, deployment)
 	Expect(err).NotTo(HaveOccurred())
+
+	framework.Logf("Ensure that both old replica sets have no replicas")
```
Comment (mfojtik) [inline test/e2e/deployment.go]: the below can be wrapped in a func ensureReplicas(n) and reused above, but I guess it is fine

File: test/e2e/deployment.go [hunk 3]
```diff
@@ -896,18 +914,22 @@ func testScalePausedDeployment(f *framework.Framework) {
 	err = framework.WaitForObservedDeployment(c, ns, deploymentName, deployment.Generation)
 	Expect(err).NotTo(HaveOccurred())
 
-	rs, err := deploymentutil.GetNewReplicaSet(deployment, c)
-	Expect(err).NotTo(HaveOccurred())
+	framework.Logf("Waiting for deployment %q to have no running pods", deploymentName)
+	Expect(framework.WaitForDeploymentUpdatedReplicasLTE(c, ns, deploymentName, replicas, deployment.Generation))
```
Comment (mfojtik) [inline test/e2e/deployment.go]: is this waiting for the deployment to have 0 replicas?

File: test/e2e/deployment.go (hunk 1 above)
Comment (kargakis) [inline test/e2e/deployment.go]: Previously the deployment was using 3 replicas and a maxUnavailable of 1. This meant that rollover could be exercised, eg.

rs-1: 3 pods
==update==>
rs-1: 2 pods, rs-2: 2 pods
==deployment blocks because rs-2 is broken==>
==new update==>
rs-1 and rs-2 are rolled over by rs-3

Now that the deployment uses 1 pod, we need maxUnavailable=0 for continuing to exercise rollover

File: test/e2e/deployment.go (hunk 2 above)
Comment (kargakis) [inline test/e2e/deployment.go]: agreed, will update

File: test/e2e/deployment.go (hunk 3 above)
Comment (kargakis) [inline test/e2e/deployment.go]: Yes, although at this point, it's a simple check.

**Comment by kargakis** [comment]
@mfojtik comments addressed. lgty?

**Comment by mfojtik** [comment]
/lgtm

thanks

**Comment by kargakis** [comment]
@k8s-bot bazel test this

**Review by janetkuo — APPROVED** [review]
